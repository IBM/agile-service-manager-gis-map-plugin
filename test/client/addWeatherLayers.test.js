// Import the module under test
import addWeatherLayers from '../../src/scripts/client/addWeatherLayers';

describe('addWeatherLayers', function() {
    test('add overlay', function() {
        const overlay = {}
        addWeatherLayers(overlay, 'myKey')
        expect(overlay).toMatchSnapshot();
    });
});
