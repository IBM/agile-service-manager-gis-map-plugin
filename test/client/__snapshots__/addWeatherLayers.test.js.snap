// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addWeatherLayers add overlay 1`] = `
Object {
  "Clouds": NewClass {
    "_events": Object {
      "tileunload": Array [
        Object {
          "ctx": undefined,
          "fn": [Function],
        },
      ],
    },
    "_initHooksCalled": true,
    "_url": "https://tile.openweathermap.org/map/clouds_new/{z}/{x}/{y}.png?appid=myKey",
    "options": Object {
      "attribution": "Weather from <a href=\\"http://openweathermap.org/\\" alt=\\"World Map and worldwide Weather Forecast online\\">OpenWeatherMap</a>",
      "maxClusterRadius": 120,
      "maxZoom": 19,
      "subdomains": Array [
        "a",
        "b",
        "c",
      ],
    },
  },
  "Precipitation": NewClass {
    "_events": Object {
      "tileunload": Array [
        Object {
          "ctx": undefined,
          "fn": [Function],
        },
      ],
    },
    "_initHooksCalled": true,
    "_url": "https://tile.openweathermap.org/map/precipitation_new/{z}/{x}/{y}.png?appid=myKey",
    "options": Object {
      "attribution": "Weather from <a href=\\"http://openweathermap.org/\\" alt=\\"World Map and worldwide Weather Forecast online\\">OpenWeatherMap</a>",
      "maxClusterRadius": 120,
      "maxZoom": 19,
      "subdomains": Array [
        "a",
        "b",
        "c",
      ],
    },
  },
  "Sea level pressure": NewClass {
    "_events": Object {
      "tileunload": Array [
        Object {
          "ctx": undefined,
          "fn": [Function],
        },
      ],
    },
    "_initHooksCalled": true,
    "_url": "https://tile.openweathermap.org/map/pressure_new/{z}/{x}/{y}.png?appid=myKey",
    "options": Object {
      "attribution": "Weather from <a href=\\"http://openweathermap.org/\\" alt=\\"World Map and worldwide Weather Forecast online\\">OpenWeatherMap</a>",
      "maxClusterRadius": 120,
      "maxZoom": 19,
      "subdomains": Array [
        "a",
        "b",
        "c",
      ],
    },
  },
  "Temperature": NewClass {
    "_events": Object {
      "tileunload": Array [
        Object {
          "ctx": undefined,
          "fn": [Function],
        },
      ],
    },
    "_initHooksCalled": true,
    "_url": "https://tile.openweathermap.org/map/temp_new/{z}/{x}/{y}.png?appid=myKey",
    "options": Object {
      "attribution": "Weather from <a href=\\"http://openweathermap.org/\\" alt=\\"World Map and worldwide Weather Forecast online\\">OpenWeatherMap</a>",
      "maxClusterRadius": 120,
      "maxZoom": 19,
      "subdomains": Array [
        "a",
        "b",
        "c",
      ],
    },
  },
  "Wind speed": NewClass {
    "_events": Object {
      "tileunload": Array [
        Object {
          "ctx": undefined,
          "fn": [Function],
        },
      ],
    },
    "_initHooksCalled": true,
    "_url": "https://tile.openweathermap.org/map/wind_new/{z}/{x}/{y}.png?appid=myKey",
    "options": Object {
      "attribution": "Weather from <a href=\\"http://openweathermap.org/\\" alt=\\"World Map and worldwide Weather Forecast online\\">OpenWeatherMap</a>",
      "maxClusterRadius": 120,
      "maxZoom": 19,
      "subdomains": Array [
        "a",
        "b",
        "c",
      ],
    },
  },
}
`;
